pkg/transform/kubernetes/kubernetes.go

```diff
@@ -416,32 +420,36 @@ func CreateK8sObjects(app *spec.App) ([]runtime.Object, error) {
 	objects = append(objects, pvcs...)
 	log.Debugf("app: %s, pvc: %s\n", app.Name, spew.Sprint(pvcs))
 
-	return objects, nil
+	return objects, app.ExtraResources, nil
 }
 
-func Transform(app *spec.App) ([]runtime.Object, error) {
+// Transform function if given spec.App data creates the versioned
+// kubernetes objects and returns them in list of runtime.Object
+// And if the field in spec.App called 'extraResources' is used
+// then it returns the filenames mentioned there as list of string
+func Transform(app *spec.App) ([]runtime.Object, []string, error) {
 
-	runtimeObjects, err := CreateK8sObjects(app)
+	runtimeObjects, extraResources, err := CreateK8sObjects(app)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to create Kubernetes objects")
+		return nil, extraResources, errors.Wrap(err, "failed to create Kubernetes objects")
```

Comment:
umm, any reason we are returning `return nil, extraResources, "error"` and not `return nil, nil, "error")` here and everywhere else?